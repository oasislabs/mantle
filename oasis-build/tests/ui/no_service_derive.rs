use oasis_std::Context;

pub struct Counter(u32);

impl Counter {
    pub fn new(ctx: &Context) -> Self {
        Self(42)
    }
}

fn main() {
    oasis_std::service!(Counter);
}
