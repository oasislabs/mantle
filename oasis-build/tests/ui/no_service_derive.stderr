error[E0599]: no function or associated item named `sunder` found for type `Counter` in the current scope
  --> /tmp/oasis_generated/no_service_derive_ctor-0a5ed9a99cf4b401.rs:10:16
   |
10 |     <Counter>::sunder(service);
   |                ^^^^^^ function or associated item not found in `Counter`
   | 
  ::: $DIR/no_service_derive.rs:3:1
   |
3  | pub struct Counter(u32);
   | ------------------------ function or associated item `sunder` not found for this
   |
   = help: items from traits can only be used if the trait is implemented and in scope
   = note: the following trait defines an item `sunder`, perhaps you need to implement it:
           candidate #1: `oasis_std::exe::Service`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0599`.
